src/workerd/api/html-rewriter.c++

```diff
@@ -754,15 +758,12 @@ void Rewriter::outputImpl(kj::ArrayPtr<const byte> buffer) {
     return;
   }
 
-  auto bufferCopy = kj::heapArray(buffer);
-
-  KJ_IF_SOME(wp, writePromise) {
-    writePromise = wp.then([this, bufferCopy = kj::mv(bufferCopy)]() mutable {
-      return inner->write(bufferCopy.asPtr()).attach(kj::mv(bufferCopy));
-    });
-  } else {
-    writePromise = inner->write(bufferCopy.asPtr()).attach(kj::mv(bufferCopy));
+  if (outputBuffer.size() + buffer.size() > MAX_OUTPUT_BUFFER_SIZE) {
+    maybePoison(KJ_EXCEPTION(FAILED, kj::str(JSG_EXCEPTION(RangeError) ": Memory limit exceeded")));
```

Comment:
If this exceptional code path still exists after resolving @danlapid's comment about memory accounting (https://github.com/cloudflare/workerd/pull/3776/files#r2008768150), it'd be nice to have a test which exercises it. That is, a test case which expects this RangeError to be thrown.